lib/Jsrt/Jsrt.cpp

```diff
@@ -1480,7 +1483,16 @@ CHAKRA_API JsHasProperty(_In_ JsValueRef object, _In_ JsPropertyIdRef propertyId
         *hasProperty = Js::JavascriptOperators::OP_HasProperty(object, ((Js::PropertyRecord *)propertyId)->GetPropertyId(), scriptContext) != 0;
 
         return JsNoError;
-    });
+    };
+
+    if (Js::JavascriptProxy::Is(object))
+    {
+        return ContextAPIWrapper<JSRT_MAYBE_TRUE>(internalHasProperty);
+    }
+    else
+    {
+        return ContextAPINoScriptWrapper(internalHasProperty);
```

Comment:
Added prototype checking. (saw no harm on performance) @kunalspathak thanks for this example and chat on getter/setter. Current implementation, couldn't trigger JS code while inside a ContextAPINoScriptWrapper. Additional concerns welcome!